Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-178.ca-central-1.compute.internal
Sha: 26dc541ecb4de118367e06f37adbca3d5bcb473c
Number of Error(s): 19
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
CMake Error at /home/centos/smoketest/PR-14844/build/generate-boost-regex-prefix/src/generate-boost-regex-stamp/download-generate-boost-regex.cmake:159 (message):
    error: downloading https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz failed
  The requested URL returned error: 403 Forbidden
         error: downloading https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz failed
  The requested URL returned error: 403 Forbidden
         error: downloading https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz failed
  The requested URL returned error: 403 Forbidden
         error: downloading https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz failed
  The requested URL returned error: 403 Forbidden
         error: downloading https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz failed
  The requested URL returned error: 403 Forbidden
         error: downloading https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz failed
  The requested URL returned error: 403 Forbidden
make[2]: *** [CMakeFiles/generate-boost-regex.dir/build.make:90: generate-boost-regex-prefix/src/generate-boost-regex-stamp/generate-boost-regex-download] Error 1
make[1]: *** [CMakeFiles/Makefile2:937: CMakeFiles/generate-boost-regex.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
CMake Error at /home/centos/smoketest/PR-14844/build/generate-boost-regex-prefix/src/generate-boost-regex-stamp/download-generate-boost-regex.cmake:159 (message):
  Each download failed!
    error: downloading https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz failed
         status_code: 22
         status_string: HTTP response code said error
         log:
         --- LOG BEGIN ---
           Trying 52.40.49.72...
  TCP_NODELAY set
  Connected to dl.bintray.com (52.40.49.72) port 443 (#0)
  ALPN, offering http/1.1
  successfully set certificate verify locations:
    CAfile: /etc/pki/tls/certs/ca-bundle.crt
    CApath: none
  [5 bytes data]
  TLSv1.3 (OUT), TLS handshake, Client hello (1):
  [512 bytes data]
  [5 bytes data]
  TLSv1.3 (IN), TLS handshake, Server hello (2):
  [102 bytes data]
  NPN, negotiated HTTP1.1
  [5 bytes data]
  TLSv1.2 (IN), TLS handshake, Certificate (11):
  [2765 bytes data]
  [5 bytes data]
  TLSv1.2 (IN), TLS handshake, Server key exchange (12):
  [333 bytes data]
  [5 bytes data]
  TLSv1.2 (IN), TLS handshake, Server finished (14):
  [4 bytes data]
  [5 bytes data]
  TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
  [70 bytes data]
  [5 bytes data]
  TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
  [1 bytes data]
  [5 bytes data]
  TLSv1.2 (OUT), TLS handshake, Next protocol (67):
  [36 bytes data]
  [5 bytes data]
  TLSv1.2 (OUT), TLS handshake, Finished (20):
  [16 bytes data]
  [5 bytes data]
  [5 bytes data]
  TLSv1.2 (IN), TLS handshake, Finished (20):
  [16 bytes data]
  SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
  ALPN, server did not agree to a protocol
  Server certificate:
   subject: CN=*.bintray.com
   start date: Sep 26 00:00:00 2019 GMT
   expire date: Nov  9 12:00:00 2021 GMT
   subjectAltName: host dl.bintray.com matched cert\s *.bintray.com
   issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=GeoTrust RSA CA 2018
   SSL certif
 ... ( comment is too long, 1198 lines, 100759 bytes, truncated) ... 
[Build error:](https://gist.github.com/b292bfb4deb4173337e623e672cb05e8)
Build log [Build log](https://gist.github.com/HPCCSmoketest/b292bfb4deb4173337e623e672cb05e8#file-result-21-04-12-15-09-23-log)

```

--- a/thorlcr/activities/thdiskbaseslave.cpp
+++ b/thorlcr/activities/thdiskbaseslave.cpp
@@ -414,33 +414,40 @@ void CDiskWriteSlaveActivityBase::close()
 {
     try
     {
-        if (out) {
-            uncompressedBytesWritten = out->getPosition();
-            if (calcFileCrc) {
-                if (diskHelperBase->getFlags() & TDWextend) {
-                    assertex(!"TBD need to merge CRC");
-                }   
-                else
-                    out->flush(&fileCRC);
-            }
-            else if (!abortSoon)
-                out->flush();
-            out.clear();
-        }
-        else if (outraw) {
-            outraw->flush();
-            uncompressedBytesWritten = outraw->tell();
-            outraw.clear();
-        }
-
-        Owned<IFileIO> tmpFileIO;
+        if (outputIO)
         {
-            CriticalBlock block(outputCs);
-            // ensure it is released/destroyed after releasing crit, since the IFileIO might involve a final copy and take considerable time.
-            tmpFileIO.setown(outputIO.getClear());
+            if (out)
+            {
+                uncompressedBytesWritten = out->getPosition();
+                if (calcFileCrc)
+                {
+                    if (diskHelperBase->getFlags() & TDWextend)
+                    {
+                        assertex(!"TBD need to merge CRC");
+                    }
+                    else
+                        out->flush(&fileCRC);
+                }
+                else if (!abortSoon)
+                    out->flush();
+                out.clear();
+            }
+            else if (outraw)
+            {
+                outraw->flush();
+                uncompressedBytesWritten = outraw->tell();
+                outraw.clear();
+            }
+
+            Owned<IFileIO> tmpFileIO;
+            {
+                CriticalBlock block(outputCs);
+                // ensure it is released/destroyed after releasing crit, since the IFileIO might involve a final copy and take considerable time.
+                tmpFileIO.setown(outputIO.getClear());
+            }
+            mergeStats(stats, tmpFileIO, diskWriteRemoteStatistics);
+            tmpFileIO->close(); // NB: close now, do not rely on close in dtor
         }
-        mergeStats(stats, tmpFileIO, diskWriteRemoteStatistics);
-        tmpFileIO->close(); // NB: close now, do not rely on close in dtor
 
         if (!rfsQueryParallel && dlfn.isExternal() && !lastNode())
         {

